addons/website/models/res_users.py

```diff
@@ -95,3 +95,10 @@ def authenticate(cls, db, login, password, user_agent_env):
                 visitor_pre_authenticate_sudo.access_token = user_partner.id
                 visitor_pre_authenticate_sudo._update_visitor_last_visit()
         return uid
+
+    @api.constrains('groups_id')
+    def _check_one_user_type(self):
+        super()._check_one_user_type()
+        internal_users = self.env.ref('base.group_user').users & self
```

Comment:
why `& self`

Reply:
to do the intersection of `self.env.ref('base.group_user').users` and `self` as you proposed [here](https://github.com/odoo/odoo/pull/121156#discussion_r1585084279). That way if there is an existing internal user with a website id, we don't trigger the error when the admin change another user.